src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -382,12 +385,18 @@ private IDataView GetShuffledData(string path)
             metrics.Train = new TrainMetrics();
             float accuracy = 0;
             float crossentropy = 0;
+            TrainState trainstate = new TrainState();
+            trainstate.BatchSize = options.BatchSize;
+            trainstate.BatchesPerEpoch = (int)(trainingSet.GetRowCount() ?? 0 / options.BatchSize);
```

Comment:
>trainstate.BatchesPerEpoch = (int)(trainingSet.GetRowCount() ?? 0 / options.BatchSize); [](start = 12, length = 87)

GetRowCount() can be null even when there are rows, so this line will result in an incorrect number of batches per epoch. If total number of examples/rows isn't available anywhere then you can simple calculate that in CacheFeaturizedImagesToDisk as we are featuring every image by keeping a counter and then passing that counter to this method.